Testscripts/Linux/XDPForwardingTest.sh

```diff
@@ -0,0 +1,181 @@
+    ssh $forwarder $commandMACS
+    commandMACD="sed -i 's/unsigned char newethdest \[\] = { 0x00, 0x22, 0x48, 0x4c, 0xc0, 0xfd };/unsigned char newethdest \[\] = { ${receiverMAC1} };/g' ${xdpdumpFileName}"
+    ssh $forwarder $commandMACD
+    LogMsg "Updated Source &  Destination MAC address in $xdpdumpFileName on $forwarder"
+    commandIPS="sed -i 's/__u8 newsrc \[\] = { 10, 0, 1, 5 };/__u8 newsrc \[\] = { ${forwarderIP1} };/g' ${xdpdumpFileName}"
+    ssh $forwarder $commandIPS
+    commandIPD="sed -i 's/__u8 newdest \[\] = { 10, 0, 1, 4 };/__u8 newdest \[\] = { ${receiverIP1} };/g' ${xdpdumpFileName}"
+    ssh $forwarder $commandIPD
+    LogMsg "Updated Source &  Destination IP address in $xdpdumpFileName on $forwarder"
+}
+
+UTIL_FILE="./utils.sh"
+
+# Source utils.sh
+. ${UTIL_FILE} || {
+    echo "ERROR: unable to source ${UTIL_FILE}!"
+    echo "TestAborted" > state.txt
+    exit 0
+}
+
+XDPUTIL_FILE="./XDPUtils.sh"
+
+# Source XDPUtils.sh
+. ${XDPUTIL_FILE} || {
+    LogMsg "ERROR: unable to source ${XDPUTIL_FILE}!"
+    SetTestStateAborted
+    exit 0
+}
+
+# Source constants file and initialize most common variables
+UtilsInit
+# Script start from here
+LogMsg "*********INFO: Script execution Started********"
+LogMsg "forwarder : ${forwarder}"
+LogMsg "receiver : ${receiver}"
+LogMsg "nicName: ${nicName}"
+bash ./XDPDumpSetup.sh ${forwarder} ${nicName}
+check_exit_status "XDPDumpSetup on ${forwarder}" "exit"
+SetTestStateRunning
+bash ./XDPDumpSetup.sh ${receiver} ${nicName}
+check_exit_status "XDpDUMPSetup on ${receiver}" "exit"
+SetTestStateRunning
+configure_XDPDUMP_TX
+
+LogMsg "XDP Setup Completed"
+
+# Setup pktgen on Sender
+LogMsg "Configure pktgen on ${sender}"
+pktgenDir=~/pktgen
+ssh ${sender} "mkdir -p ${pktgenDir}"
+download_pktgen_scripts ${sender} ${pktgenDir} ${cores}
+# Configure XDP_TX on Forwarder
+LogMsg "Build XDPDump with TX Action on ${forwarder}"
+ssh ${forwarder} "cd bpf-samples/xdpdump && make clean && CFLAGS='-D __TX_FWD__ -D __PERF__ -I../libbpf/src/root/usr/include' make"
+check_exit_status "Build xdpdump with TX Action on ${forwarder}"
+# Configure XDP_DROP on receiver
+LogMsg "Build XDPDump with DROP Action on ${receiver}"
+ssh ${receiver} "cd bpf-samples/xdpdump && make clean && CFLAGS='-D __PERF_DROP__ -D __PERF__ -I../libbpf/src/root/usr/include' make"
+check_exit_status "Build xdpdump with DROP Action on ${receiver}"
+
+# Calculate packet drops before tests
+packetDropBefore=$(ssh ${receiver} ". XDPUtils.sh && calculate_packets_drop ${nicName}")
+LogMsg "Before test, Packet drop count on ${receiver} is ${packetDropBefore}"
+# Calculate packets forwarded before tests
+pktForwardBefore=$(ssh ${forwarder} ". XDPUtils.sh && calculate_packets_forward ${nicName}")
+LogMsg "Before test, Packet forward count on ${forwarder} is ${pktForwardBefore}"
+
+# Start XDPDump on receiver
```

Comment:
a few more other places has similar 'dup' code, not exact the same, but 90% codes are same, could we wrap with some shared function as wrapper?

Reply:
fixed